raft/src/main/java/org/apache/kafka/raft/ReplicatedLog.java

```diff
@@ -30,14 +30,14 @@
      *
      * @return the base offset of the first appended record.
      */
-    Long appendAsLeader(Records records, int epoch);
+    LogAppendInfo appendAsLeader(Records records, int epoch);
```

Comment:
I'd need to return at least the base and last offset and hence introduced a new struct here.